crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -152,26 +152,34 @@ def adminrepocheck
           os_available = repo_version_available?(products, "SLES", "12.3")
           ret[:os] = {
             available: os_available,
-            repos: {}
+            repos: ::Crowbar::Repository.feature_repository_map(Api::Upgrade.target_platform)["os"],
+            errors: {}
           }
-          ret[:os][:repos][admin_architecture.to_sym] = {
-            missing: ["SLES-12-SP3-Pool", "SLES12-SP3-Updates"]
-          } unless os_available
+          unless os_available
+            ret[:os][:errors][admin_architecture.to_sym] = {
+              missing: ret[:os][:repos]
+            }
+          end
 
           cloud_available = repo_version_available?(products, "suse-openstack-cloud", "8")
           ret[:openstack] = {
             available: cloud_available,
-            repos: {}
+            repos: ::Crowbar::Repository.feature_repository_map(
```

Comment:
Admin repochecks are triggerend on the "old cloud" side so the repos defined in the system will not reflect what we want to see for the upgrade.